generators/client/templates/angular/src/main/webapp/app/account/register/register.component.ts.ejs

```diff
@@ -31,51 +32,70 @@ import { Register } from './register.service';
 })
 export class RegisterComponent implements OnInit, AfterViewInit {
 
-    confirmPassword: string;
     doNotMatch: string;
     error: string;
     errorEmailExists: string;
     errorUserExists: string;
-    registerAccount: any;
     success: boolean;
     modalRef: NgbModalRef;
 
+    registerForm = this.fb.group({
+        login: ['', [Validators.required, Validators.minLength(1), Validators.maxLength(50), Validators.pattern('^[_.@A-Za-z0-9-]*$')]],
+        email: ['', [Validators.required, Validators.minLength(5), Validators.maxLength(254), Validators.email]],
+        password: ['', [Validators.required, Validators.minLength(4), Validators.maxLength(50)]],
+        confirmPassword: ['', [Validators.required, Validators.minLength(4), Validators.maxLength(50)]]
+    });
+
     constructor(
         private languageService: JhiLanguageService,
         private loginModalService: LoginModalService,
         private registerService: Register,
         private elementRef: ElementRef,
-        private renderer: Renderer
+        private renderer: Renderer,
+        private fb: FormBuilder
     ) {
     }
 
     ngOnInit() {
         this.success = false;
-        this.registerAccount = {};
     }
 
     ngAfterViewInit() {
         this.renderer.invokeElementMethod(this.elementRef.nativeElement.querySelector('#login'), 'focus', []);
     }
 
     register() {
-        if (this.registerAccount.password !== this.confirmPassword) {
+        let registerAccount = {};
+        const login = this.registerForm.get(['login']).value;
+        const email = this.registerForm.get(['email']).value;
+        const password = this.registerForm.get(['password']).value;
+        if (password !== this.registerForm.get(['confirmPassword']).value) {
             this.doNotMatch = 'ERROR';
         } else {
+            registerAccount = Object.assign(registerAccount, {
```

Comment:
Maybe we can use spread operator here:
`registerAccount = {...registerAccount, login, email, password} `